File: API/Client/src/model/textCompletion.ts
export interface TextCompletion {
	prompt_tokens: string[]
	result_tokens: string[]
	alt_token_groups: string[][]
	alt_token_prob_groups: number[][]
}

export const EmptyCompletion: TextCompletion = {
	prompt_tokens: [],
	result_tokens: [],
	alt_token_groups: [],
	alt_token_prob_groups: []
}

export const extractGeneratedTokens = (textCompletion: TextCompletion): string[] => {
	// Remove the prompt tokens from the result tokens
	// to obtain only tokens that were generated
	return textCompletion.result_tokens.slice(textCompletion.prompt_tokens.length)
}

File: DomainEntities/TextCompletion.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace DomainEntities
{
    public class TextCompletion
    {
        [JsonPropertyName("prompt_tokens")]
        public List<string> PromptTokens { get; private set; }

        [JsonPropertyName("result_tokens")]
        public List<string> ResultTokens { get; private set; }

        [JsonPropertyName("alt_token_groups")]
        public List<List<string>> AltTokenGroups { get; private set; }

        [JsonPropertyName("alt_token_prob_groups")]
        public List<List<float>> AltTokenProbGroups { get; private set; }

        public TextCompletion(
            List<string> promptTokens,
            List<string> resultTokens,
            List<List<string>> altTokenGroups,
            List<List<float>> altTokenProbGroups)
        {
            PromptTokens = promptTokens;
            ResultTokens = resultTokens;
            AltTokenGroups = altTokenGroups;
            AltTokenProbGroups = altTokenProbGroups;
        }
    }
}
